modules/market/src/test/java/com/opengamma/strata/market/param/ParameterizedDataCombinerTest.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.param;
+
+import static com.opengamma.strata.collect.TestHelper.assertThrows;
+import static com.opengamma.strata.collect.TestHelper.assertThrowsIllegalArg;
+import static org.testng.Assert.assertEquals;
+
+import org.testng.annotations.Test;
+
+/**
+ * Test {@link ParameterizedDataCombiner}.
+ */
+@Test
+public class ParameterizedDataCombinerTest {
+
+  private static final TestingParameterizedData2 DATA1 = new TestingParameterizedData2(1d, 2d);
+  private static final TestingParameterizedData DATA2 = new TestingParameterizedData(3d);
+  private static final TestingParameterizedData2 DATA3 = new TestingParameterizedData2(4d, 5d);
+
+  //-------------------------------------------------------------------------
+  public void test_basics() {
+    ParameterizedDataCombiner test = ParameterizedDataCombiner.of(DATA1, DATA2, DATA3);
+    assertEquals(test.getParameterCount(), 5);
+    assertEquals(test.getParameter(0), 1d);
+    assertEquals(test.getParameter(1), 2d);
+    assertEquals(test.getParameter(2), 3d);
+    assertEquals(test.getParameter(3), 4d);
+    assertEquals(test.getParameter(4), 5d);
+    assertEquals(test.getParameterMetadata(0), ParameterMetadata.empty());
+    assertThrows(() -> test.getParameter(-1), IndexOutOfBoundsException.class);
+    assertThrows(() -> test.getParameter(5), IndexOutOfBoundsException.class);
+    assertThrows(() -> test.getParameterMetadata(-1), IndexOutOfBoundsException.class);
+    assertThrows(() -> test.getParameterMetadata(5), IndexOutOfBoundsException.class);
+    assertThrowsIllegalArg(() -> ParameterizedDataCombiner.of());
```

Comment:
I see the test with empty (size = 0) underlyings, but can we add tests with one or more null underlyings. I assume that should also throw an illegal arg exception.